openeo/local/connection.py

```diff
@@ -282,4 +285,16 @@ def execute(
         if auto_decode is not True:
             raise ValueError("LocalConnection requires auto_decode=True")
         process_graph = as_flat_graph(process_graph)
-        return OpenEOProcessGraph(process_graph).to_callable(PROCESS_REGISTRY)()
+        pg = OpenEOProcessGraph(process_graph)
+
+        # Return the result and get the workflow provinance (yprov4wfs)
+        result = pg.to_callable(PROCESS_REGISTRY)()
+        workflow = pg.workflow
+
+        # To save the provenance file in the specific path use:
+        # workflow.prov_to_json(directory_path=save_path)
+
+        if return_provenance:
+            return result, workflow
```

Comment:
This custom return should be documented in the docblock and return annotation

Reply:
that being said, I'm not a big fan of the pattern of returning different data structures (tuple of two things instead of a single `DataArray`) depending on input arguments.
Especially, because in normal usage of the openEO python client, the `execute` method of connection objects (`LocallConnection` here) is usually not used directly by users, but indirectly through `DataCube.execute()` or something equivalent. So changing the input and output API of `Connection.execute` is going to create problems